pkg/controller/render/render_controller.go

```diff
@@ -563,7 +563,16 @@ func generateRenderedMachineConfig(pool *mcfgv1.MachineConfigPool, configs []*mc
 		}
 	}
 
-	merged, err := ctrlcommon.MergeMachineConfigs(configs, cconfig.Spec.OSImageURL)
+	// If our new image formats are populated
+	// TODO(jkyros): "properly" feature gate this? This is kind of a 'soft' gate based on the presence of extensions container,
```

Comment:
Ah, OK.  That is indeed one way to get it tested in CI.

That said it also has the effect that CI and nightlies will behave differently...so I think I'd vote for a `const enableNewImage = false` that can be flipped on.  Or we just do the more proper feature gating via machineconfig.